Tests added

EquinoxBlack opened the pull request.

mmartinadanx commented on integration-tests/pom.xml:
```diff
@@ -0,0 +1,63 @@
+  </properties>
+
+  <dependencies>
+<!--    <dependency>-->
```
For now, I'd remove this code instead of commenting it.

EquinoxBlack replied:
removed

mmartinadanx commented on integration-tests/src/test/java/org/alfresco/core/SitesApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,120 @@
+public class SitesApiServiceIntegrationTest {
+
+    @Container
+    public static DockerComposeContainer compose = new DockerComposeContainer(new File("src/test/resources/docker-compose.yml"))
```
These literal values like: 
* Name of the service
* Port
* Duration in seconds
* Poll delay
Would it be OK to promote them to constant or config properties?

mmartinadanx commented on integration-tests/src/test/resources/docker-compose.yml:
```diff
@@ -0,0 +1,129 @@
+services:
+
+  alfresco:
+    image: alfresco/alfresco-content-repository-community:7.0.0-A29
```
Use the latest released version.

EquinoxBlack replied:
updated with the latest tag

mmartinadanx commented on integration-tests/src/test/java/org/alfresco/core/NodeApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    void should_createNode_and_getNode() {
```
@zoltanpalfi are we finally OK with this naming?. I.e. When we're only testing the happy path, we don't use "_when_" and the test outcome is the name of the method we're testing. If so, should we add this to our code convention wiki?

mmartinadanx reviewed (commented):
Well done! A few comments for discussion.

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/CommentsApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    @Test
+    void should_createComment_and_listComments() {
```
Naming of the test doesn't confirm to our convention. There should be no `_and_` with underline, rather an `And`

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/DownloadsApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Test
+    void should_createDownload_and_getDownload() {
```
Naming doesn't confirm to test convention.

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/DownloadsApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,106 @@
+        long awaitTimeoutInMilliseconds = 120000;
+        await()
+            .atMost(awaitTimeoutInMilliseconds, TimeUnit.MILLISECONDS)
+            .pollDelay(100000, TimeUnit.MILLISECONDS)
```
a nicer way is to separate with underscores the thousands and millions, so `100_000`

EquinoxBlack replied:
done

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/DownloadsApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,106 @@
+            .pollDelay(100000, TimeUnit.MILLISECONDS)
+            .untilAsserted(() ->
+                assertThat(nodesApiClient
+                    .getNode("-my-", null, null, null)
```
`-my-` should be converted to a private static constant

EquinoxBlack replied:
done

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/GroupsApiServiceIntegrationTests.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Test
+    void should_createGroup_and_getGroup() {
```
test naming not conforming to convention

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/NetworksApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,82 @@
+
+    @BeforeAll
+    public void setUp() {
+        long awaitTimeoutInMilliseconds = 120000;
```
could be improve if you agree this is more readable: `2 * 60 * 1000`

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/NetworksApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,82 @@
+        long awaitTimeoutInMilliseconds = 120000;
+        await()
+            .atMost(awaitTimeoutInMilliseconds, TimeUnit.MILLISECONDS)
+            .pollDelay(100000, TimeUnit.MILLISECONDS)
```
separation of thousands improves readability: `100_000`

EquinoxBlack replied:
done

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/NetworksApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,82 @@
+            .pollDelay(100000, TimeUnit.MILLISECONDS)
+            .untilAsserted(() ->
+                assertThat(nodesApiClient
+                    .getNode("-my-", null, null, null)
```
-my- => Constant, maybe it's even worth to have a CommonConstants class in tests where you have it as public

EquinoxBlack replied:
done

zoltanpalfi commented on integration-tests/src/test/java/org/alfresco/core/NodesApiServiceIntegrationTest.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    void should_createNode_and_getNode() {
```
test naming here too

zoltanpalfi reviewed (changes requested):
Good progress on the tests! 👍 
Please follow what we have agreed on test naming convention: either `should_doSomething` or `should_doSomething_when_someConditions`.
Test should verify one behaviour but I understand occasionally it may be required to check something in the same test. In that case the correct usage of our convention is `should_doSomethingAndSomeMore` or `should_doSomethingAndSomeMore_when_someCondition`.
I flagged a few, but there are plenty of these, please apply to all. We should start focusing on merging now, this looks good and if we continue to add more and more, will overgrow and review will become much harder.

EquinoxBlack committed:
Integration tests updated

zoltanpalfi reviewed (approved):
Looks good

josorno-ccc committed:
Merge branch 'develop' into dev-rodion-REPO-5614

josorno-ccc committed:
REPO-5614 New integration tests approach

Test containers approach using local docker environment made the tests
too slow, so we have decided to change the approach and use an existing
pre-allocated environment as the target of the tests.

The target environment properties must be set using environment variables
in each case.

josorno-ccc committed:
REPO-5614 Remove unrequired config for Travis

josorno-ccc committed:
REPO-5614 Improve tests

josorno-ccc committed:
REPO-5614 Ensure comment creation

josorno-ccc committed:
REPO-5614 Remove unnecessary code

josorno-ccc committed:
REPO-5614 Improve find sites test

josorno-ccc committed:
REPO-5614 Update readme file

josorno-ccc committed:
REPO-5614 Improve comments test class reliability
